### PR TITLE
Claimable fees and removable liquidity

### PR DESCRIPTION
**Pull request by fpieper**

### CODE REVIEW

File: src/pool.rs [hunk 1]
```diff
@@ -798,6 +800,53 @@ mod precision_pool {
...
+            let mut y_total_output = dec!(0);
+
+            for position_id in lp_position_ids {
+                let position: LiquidityPosition =
+                    self.lp_manager.get_non_fungible_data(&position_id);
+                let price_left_sqrt = tick_to_price_sqrt(position.left_bound);
+                let price_right_sqrt = tick_to_price_sqrt(position.right_bound);
+
+                let (x_fees, y_fees, _, _) = self.claimable_fees_internal(&position);
+                let (x_amount, y_amount) = remove_amounts(
```
Comment (flix59) [inline src/pool.rs]: looks like fees are getting removed here

File: src/pool.rs [hunk 2]
```diff
@@ -1312,13 +1422,11 @@ mod precision_pool {
...
         /// # Arguments
-        /// * `position_id` - The identifier of the non-fungible token representing the liquidity position.
+        /// * `position` - A reference to the `LiquidityPosition` struct representing the liquidity position.
         ///
         /// # Returns
         /// * `(Decimal, Decimal)` - A tuple containing the accrued `x` and `y` fees.
-        pub fn total_fees(&self, position_id: NonFungibleLocalId) -> (Decimal, Decimal) {
-            let position: LiquidityPosition = self.lp_manager.get_non_fungible_data(&position_id);
-
+        fn total_fees_internal(&self, position: &LiquidityPosition) -> (Decimal, Decimal) {
```
Comment (flix59) [inline src/pool.rs:1437]:
```suggestion
        fn total_fee_internal(&self, position: &LiquidityPosition) -> (Decimal, Decimal) {
```

File: src/pool.rs [hunk 3]
```diff
@@ -1353,6 +1461,32 @@ mod precision_pool {
...
+        /// A tuple containing two `Decimal`s:
+        /// - The first `Decimal` represents the total `x` fees.
+        /// - The second `Decimal` represents the total `y` fees.
+        pub fn total_fees(&self, lp_position_ids: Vec<NonFungibleLocalId>) -> (Decimal, Decimal) {
+            let mut x_fees = dec!(0);
+            let mut y_fees = dec!(0);
+            for position_id in lp_position_ids {
+                let position: LiquidityPosition =
+                    self.lp_manager.get_non_fungible_data(&position_id);
+                let (x_total, y_total) = self.total_fees_internal(&position);
```
Comment (flix59) [inline src/pool.rs:1491]:
```suggestion
                let (x_total, y_total) = self.total_fee_internal(&position);
```

**Review by flix59 — APPROVED** [review]
looks good to me, added some minor comments

File: src/pool.rs (hunk 2 above)
Comment (fpieper) [reply src/pool.rs:1437]: I will stick with the current version because we are talking about all fees a position has earned since the beginning, but thx 👍

**Pull request merged**